server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexAliasesService.java

```diff
@@ -127,7 +127,7 @@ ClusterState innerExecute(ClusterState currentState, Iterable<AliasAction> actio
                 if (index == null) {
                     throw new IndexNotFoundException(action.getIndex());
                 }
-                NewAliasValidator newAliasValidator = (alias, indexRouting, filter) -> {
+                NewAliasValidator newAliasValidator = (alias, indexRouting, filter, writeIndex) -> {
```

Comment:
this parameter isn't used anywhere. I think it should validate writeIndex semantics are sound?